### PR TITLE
Meshing/Feature-uniform-refining

### PR DESCRIPTION
**Pull request by miguelmaso**
A small change: using ModelPart*

### CODE REVIEW

File: applications/MeshingApplication/custom_utilities/uniform_refine_utility.h [hunk 1]
```diff
@@ -218,8 +218,8 @@ class UniformRefineUtility
     std::unordered_map<IndexType,int> mNodesColorMap;
     std::unordered_map<IndexType,int> mCondColorMap;
     std::unordered_map<IndexType,int> mElemColorMap;
-    std::unordered_map<int,std::vector<std::string>> mColors;  /// Where the sub model parts IDs are stored
-    std::map<std::pair<int,int>,int> mIntersections;           /// Where the colors intersection are stored
+    std::unordered_map<int,std::vector<ModelPart*>> mColorsPointers; /// Where the sub model parts are stored
```
Comment (loumalouomega) [inline applications/MeshingApplication/custom_utilities/uniform_refine_utility.h:221]: Be careful with the raw pointers!!!!, the raw pointers are evil (potential memory leak)

Comment (loumalouomega) [reply applications/MeshingApplication/custom_utilities/uniform_refine_utility.h:221]: I recommend to use smart pointers if possible

**Review by loumalouomega — DISMISSED** [review]
Please check the raw pointers for a future modification

File: applications/MeshingApplication/custom_utilities/uniform_refine_utility.h (hunk 1 above)
Comment (miguelmaso) [reply applications/MeshingApplication/custom_utilities/uniform_refine_utility.h:221]: I want to use ModelPart::Pointer but nowadays I think it is not possible...

Comment (loumalouomega) [reply applications/MeshingApplication/custom_utilities/uniform_refine_utility.h:221]: Let me check...

**Comment by loumalouomega** [comment]
@miguelmaso Can you check if this works in your machine, in mine works, but I have my reserves creating a smart pointer from the raw pointer (at the end is the same problem):

       ModelPart::Pointer p_main_model_part(&ThisModelPart);

https://stackoverflow.com/questions/4665266/creating-shared-ptr-from-raw-pointer

**Comment by loumalouomega** [comment]
If doesn't work better to revert this one and learn to life in this wicked world with raw pointers

**Comment by miguelmaso** [comment]
This is throwing a memory error. In addition, that change is breaking mmg process. I have locally reverted the changes and tried with another implementation, but I'm getting the same memory error...

**Comment by loumalouomega** [comment]
Let's use just what you did

**Comment by RiccardoRossi** [comment]
guys, YOU CAN NOT create a smart pointer from an existing pointer...

File: kratos/tests/utilities/test_sub_model_parts_list_utility.cpp [hunk 2]
```diff
@@ -265,8 +265,8 @@ namespace Kratos
                 if (key != 0) {// NOTE: key == 0 is the MainModelPart
                     if (colors.find(key) != colors.end()) {
                         for (auto sub_model_part_name : colors[key]) {
-                            ModelPart& r_sub_model_part = SubModelPartsListUtility::GetRecursiveSubModelPart(second_model_part, sub_model_part_name);
-                            r_sub_model_part.AddNode(p_node);
+                            ModelPart::Pointer r_sub_model_part = SubModelPartsListUtility::GetRecursiveSubModelPart(second_model_part, sub_model_part_name);
```
Comment (RiccardoRossi) [inline kratos/tests/utilities/test_sub_model_parts_list_utility.cpp]: why do you need a Pointer anyway?

File: kratos/tests/utilities/test_sub_model_parts_list_utility.cpp [hunk 3]
```diff
@@ -348,7 +348,7 @@ namespace Kratos
             for (auto color : colors)
             {
                 // Check both colors have the same size
-                std::vector<ModelPart*> p_color = colors_pointers[color.first];
+                std::vector<ModelPart::Pointer> p_color = colors_pointers[color.first];
```
Comment (RiccardoRossi) [inline kratos/tests/utilities/test_sub_model_parts_list_utility.cpp]: why do you need to adquire ownership?

**Comment by loumalouomega** [comment]
We know, I ask to @miguelmaso to revert it

**Review by loumalouomega — APPROVED** [review]
After reverting re-approving

**Comment by miguelmaso** [comment]
I will use a different implementation

**Pull request closed without merging**